go_wrapper/libral.go

```diff
@@ -0,0 +1,219 @@
+package libral
+
+/*
+#cgo LDFLAGS: -fPIC -L${SRCDIR}/ -lstdc++ /usr/local/lib/libaugeas.dylib /usr/local/lib/libral.a  /usr/local/lib/libcpp-hocon.a /usr/local/lib/libyaml-cpp.a /usr/local/lib/leatherman_execution.a /usr/local/lib/leatherman_logging.a /usr/local/lib/leatherman_locale.a /usr/local/lib/leatherman_ruby.a /usr/local/lib/leatherman_dynamic_library.a /usr/local/lib/leatherman_util.a /usr/local/lib/leatherman_file_util.a /usr/local/lib/leatherman_json_container.a /usr/local/lib/libboost_system-mt.a /usr/local/lib/libboost_log-mt.a /usr/local/lib/libboost_log_setup-mt.a /usr/local/lib/libboost_thread-mt.a /usr/local/lib/libboost_date_time.a /usr/local/lib/libboost_filesystem-mt.a /usr/local/lib/libboost_chrono-mt.a /usr/local/lib/libboost_regex-mt.a /usr/local/lib/libboost_atomic-mt.a /usr/local/lib/libboost_program_options-mt.a /usr/local/lib/libboost_locale-mt.a /usr/local/lib/libiconv.a
```

Comment:
Should pull in https://github.com/parisiale/libral/pull/6

For building for other targets, Go doesn't support conditionals in the preprocessor, instead you can set build constraints in the filename etc (https://golang.org/pkg/go/build/#hdr-Build_Constraints)